src/components/Combos/Combos.jsx

```diff
@@ -0,0 +1,265 @@
+    const { teamA, teamB } = this.state;
+    return getQueryString(teamA, teamB);
+  };
+
+  handleResponse = (json) => {
+    this.setState({ queryResult: json, loading: false });
+  };
+
+  sendRequest = () => {
+    const queryString = this.buildQueryString();
+    this.setState({ loading: true }, () =>
+      fetch(`${
+        process.env.REACT_APP_API_HOST
+      }/api/explorer?sql=${encodeURIComponent(queryString)}`)
+        .then(res => res.json())
+        .then(this.handleResponse));
+  };
+
+  handleSubmit = () => {
+    const { teamA, teamB } = this.state;
+    window.history.pushState(
+      '',
+      '',
+      `?${querystring.stringify({ teamA, teamB })}`,
+    );
+    this.sendRequest();
+  };
+
+  handleCancel = () => {
+    this.setState({ loading: false });
+    window.stop();
+  }
+
+  filterAndRenderElements = (searchValue, resetSearchValue) => {
+    const filteredHeroesArray = heroesArray.filter(hero =>
+      (searchValue
+        ? new RegExp(searchValue, 'i').test(hero.localized_name)
+        : true));
+
+    return [
+      ...(filteredHeroesArray.length > 0 ? filteredHeroesArray : heroesArray),
+    ].map(hero => (
+      <HeroSelector
+        id={hero.id}
+        heroName={hero.localized_name}
+        handleHeroSelection={this.handleHeroSelection(resetSearchValue)}
+        selected={[...this.state.teamA, ...this.state.teamB].includes(hero.id.toString())}
+        teamAFull={this.state.teamA.length > 4}
+        teamBFull={this.state.teamB.length > 4}
+        strings={this.props.strings}
+      />
+    ));
+  };
+
+  render() {
+    const { teamA, teamB } = this.state;
+    const { strings } = this.props;
+    return (
+      <StyledCombos>
+        <Heading title={strings.combos_title} subtitle={strings.combos_description} />
+        <HorizontalMenu
+          filterAndRenderElements={this.filterAndRenderElements}
+          filterText={strings.placeholder_filter_heroes}
+        />
+        <SelectedHeroes
+          teamA={teamA}
+          teamB={teamB}
+          handleHeroDeSelection={this.handleHeroDeSelection}
+          strings={strings}
+        />
+        <RaisedButton
+          label={this.state.loading ? strings.explorer_cancel_button : strings.request_submit}
+          onClick={this.state.loading ? this.handleCancel : this.handleSubmit}
+          style={{ marginBottom: 10 }}
+          buttonStyle={{ backgroundColor: this.state.loading ? '#822e2e' : 'rgba(23, 59, 90, 0.8)' }}
+        />
+        <pre style={{ color: 'red' }}>{this.state.queryResult.err}</pre>
+        <Heading
+          title={strings.explorer_results}
+          subtitle={`${
+            (this.state.queryResult.rows || []).length
+          } ${strings.explorer_num_rows}`}
+        />
+        {this.state.loading ? (
+          <TableSkeleton />
+        ) : (
+          <ExplorerOutputSection
+            rows={this.state.queryResult.rows}
+            fields={this.state.queryResult.fields}
```

Comment:
it would also be nice to show whether team A or team B won that match (can use the trophy icon from matches page)